Sometimes raises are not called

As of #58, there is a relatively rare bug (1 in ~1000 games?) that is caught from the fuzz tests. Here is the log:

```
[ Info: ********************************
[ Info: ******************************** Playing Game!
[ Info: ********************************
[ Info: Initial bank roll summary: (200.0, 200.0, 200.0)
[ Info: Bot5050[2] paid the small blind and dealt cards: (J♠, K♠)
[ Info: Bot5050[3] paid the  big  blind and dealt cards: (5♢, 6♢)
[ Info: Bot5050[1] dealt (free) cards:                   (7♣, 7♡)
[ Info: Table cards dealt (face-down).
[ Info: Bot5050[1] called 2.0 (blind).
[ Info: Bot5050[2] called 1.0 (blind).
[ Info: Bot5050[3] checked!
[ Info: Flop:                                              (9♡, 3♢, 4♣)
[ Info: Bot5050[1] checked!
[ Info: Bot5050[2] checked!
[ Info: Bot5050[3] raised to 83.0.
[ Info: Bot5050[1] called 83.0.
[ Info: Bot5050[2] raised to 166.0.
[ Info: Bot5050[3] folded!
[ Info: Turn:                                              J♡
[ Info: Bot5050[1] checked!
[ Info: Bot5050[2] raised to 18.0.
[ Info: Bot5050[1] raised to 61.0.
[ Info: Bot5050[2] called 14.0 (now all-in).
[ Info: River:                                             4♠
Game: Play (3 Bot5050's): Error During Test at /Users/runner/work/TexasHoldem.jl/TexasHoldem.jl/test/fuzz_play.jl:6
  Got exception outside of a @test
  AssertionError: amt_remaining ≈ 0
  Stacktrace:
   [1] contribute!(::Table, ::Player{Bot5050}, ::Float64, ::Bool) at /Users/runner/work/TexasHoldem.jl/TexasHoldem.jl/src/transactions.jl:137
   [2] raise_to_valid_raise_amount!(::Table, ::Player{Bot5050}, ::Float64) at /Users/runner/work/TexasHoldem.jl/TexasHoldem.jl/src/player_actions.jl:186
   [3] raise_to! at /Users/runner/work/TexasHoldem.jl/TexasHoldem.jl/src/player_actions.jl:180 [inlined]
   [4] raise_to!(::Game, ::Player{Bot5050}, ::Float64) at /Users/runner/work/TexasHoldem.jl/TexasHoldem.jl/src/player_actions.jl:178
   [5] player_option!(::Game, ::Player{Bot5050}, ::River, ::CheckRaiseFold) at /Users/runner/work/TexasHoldem.jl/TexasHoldem.jl/src/player_options.jl:130
   [6] player_option!(::Game, ::Player{Bot5050}) at /Users/runner/work/TexasHoldem.jl/TexasHoldem.jl/src/player_options.jl:28
   [7] act_generic!(::Game, ::River) at /Users/runner/work/TexasHoldem.jl/TexasHoldem.jl/src/game.jl:106
   [8] act! at /Users/runner/work/TexasHoldem.jl/TexasHoldem.jl/src/game.jl:112 [inlined]
   [9] play(::Game) at /Users/runner/work/TexasHoldem.jl/TexasHoldem.jl/src/game.jl:149
   [10] top-level scope at /Users/runner/work/TexasHoldem.jl/TexasHoldem.jl/test/fuzz_play.jl:8
   [11] top-level scope at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.4/Test/src/Test.jl:1113
   [12] top-level scope at /Users/runner/work/TexasHoldem.jl/TexasHoldem.jl/test/fuzz_play.jl:7
```

It seems that the first issue here is that the `Bot5050[2]` raise to `166.0.` is not called by `Bot5050[1]`. This demos something we can check for: if all BRs are equal to start, then any all-in should lead to everyone (still playing) being all in.

--- a/src/player_actions.jl
+++ b/src/player_actions.jl
@@ -70,6 +70,7 @@ function call_valid_amount!(table::Table, player::Player, amt::Real)
     @debug "$(name(player)) calling $(amt)."
     push!(player.action_history, Call(amt))
     player.action_required = false
+    player.checked = false
     blind_str = is_blind_call(table, player, amt) ? " (blind)" : ""
     contribute!(table, player, amt, true)
     if all_in(player)
@@ -189,6 +190,7 @@ function raise_to_valid_raise_amount!(table::Table, player::Player, amt::Real)
     push!(player.action_history, Raise(amt))
     player.action_required = false
     player.last_to_raise = true
+    player.checked = false
     players = players_at_table(table)
     for oponent in players
         oponent.id == player.id && continue

--- a/src/player_options.jl
+++ b/src/player_options.jl
@@ -17,16 +17,18 @@ function player_option!(game::Game, player::Player)
         if bank_roll(player) > call_amt # raise possible
             vrb = valid_raise_bounds(table, player)
             if first(vrb) ≈ last(vrb) # only all-in raise possible
-                player_option!(game, player, game_state, CallAllInFold())
+                option = CallAllInFold()
             else
-                player_option!(game, player, game_state, CallRaiseFold())
+                option = CallRaiseFold()
             end
         else # only all-in possible
-            player_option!(game, player, game_state, CallFold())
+            option = CallFold()
         end
     else
-        player_option!(game, player, game_state, CheckRaiseFold())
+        option = CheckRaiseFold()
     end
+    @debug "option = $option"
+    player_option!(game, player, game_state, option)
 end
 
 #####

--- a/test/game.jl
+++ b/test/game.jl
@@ -42,10 +42,12 @@ end
 
     # Round 3
     raise_to!(game, players[1], 10)
+    @test TH.checked(players[1]) == false
     call!(game, players[2])
 
     # Round 4
     raise_to!(game, players[1], 20)
+    @test TH.checked(players[1]) == false
     fold!(game, players[2])
 
     # All-in cases
@@ -62,6 +64,7 @@ end
 
     # Round 2
     raise_to!(game, players[1], players[1].bank_roll)
+    @test TH.checked(players[1]) == false
     call!(game, players[2])
 
     @test_throws AssertionError raise_to!(game, players[1], 10000) # raise exceeds bank roll!

